### PR TITLE
68-load-current-song-to-audio-context

### PR DESCRIPTION
**Pull request by Prom1s1ngYoung**
### Descriptions
After server-side rendering, now comes customer-side rendering, get the current index song in the music list, and load it to audio context.

- Link to UI design (if possible):

### Provide screenshots of no warning, or error in the console.
image

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: src/layouts/MusicPlayer.tsx [hunk 1]
```diff
@@ -61,6 +64,19 @@ const MusicPlayer = () => {
     }
   };
 
+  const loadSong = async () => {
+    if (playingQueue && playingQueue.songs[playingIndex]) {
+      const url = playingQueue.songs[playingIndex].mp3Url;
+      const res = await loadAudio({
```
Comment (tatoMa) [inline src/layouts/MusicPlayer.tsx:70]: Consider renaming these 2 variables. loadSong and loadAudio.

File: src/layouts/MusicPlayer.tsx [hunk 2]
```diff
@@ -61,6 +64,19 @@ const MusicPlayer = () => {
     }
   };
 
+  const loadSong = async () => {
+    if (playingQueue && playingQueue.songs[playingIndex]) {
+      const url = playingQueue.songs[playingIndex].mp3Url;
```
Comment (tatoMa) [inline src/layouts/MusicPlayer.tsx:69]: Do you need to have any conditions if an index or URL is not found?

**Review by tatoMa — COMMENTED** [review]
How about rewriting all of your related logic into a custom hook?

File: src/layouts/MusicPlayer.tsx (hunk 2 above)
Comment (Prom1s1ngYoung) [reply src/layouts/MusicPlayer.tsx:69]: I think it is needed. I will fix it.

File: src/layouts/MusicPlayer.tsx (hunk 1 above)
Comment (Prom1s1ngYoung) [reply src/layouts/MusicPlayer.tsx:70]: Sure.

**Comment by Prom1s1ngYoung** [comment]
The according logic is now been extracted to custome hooks

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request merged**